pkgs/build-support/setup-hooks/make-wrapper.sh

```diff
@@ -68,11 +70,17 @@ makeWrapper() {
             n=$((n + 1))
             flagsBefore="$flagsBefore $flags"
         fi
+
+        if test "$p" = "--argv0"; then
+            argv0=${params[$((n + 1))]}
+            n=$((n + 1))
+        fi
     done
 
     # Note: extraFlagsArray is an array containing additional flags
     # that may be set by --run actions.
-    echo exec -a '"$0"' "$original" $flagsBefore '"${extraFlagsArray[@]}"' '"$@"' >> $wrapper
+    echo exec ${argv0:+-a} $argv0 "$original" \
```

Comment:
I don't understand the `$argv0` conditional here. If argv0 is empty, it will exec `"" $original`, right? Shouldn't `-a` be there always?